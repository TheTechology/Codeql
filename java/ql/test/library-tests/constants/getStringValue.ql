import semmle.code.java.Variable

from Variable v, CompileTimeConstantExpr init, RefType enclosing, string constant
where
  v.getInitializer() = init and
  init.getEnclosingCallable().getDeclaringType() = enclosing and
  enclosing.hasQualifiedName("constants", "Values") and
  constant = init.getStringValue()
select init, constant
