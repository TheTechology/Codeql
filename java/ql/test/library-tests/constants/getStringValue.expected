| constants/Values.java:19:29:19:31 | '*' | * |
| constants/Values.java:91:37:91:45 | "a" + "b" | ab |
| constants/Values.java:92:35:92:44 | ... + ... | abc |
